velox/type/LongDecimal.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/dynamic.h>
+#include <sstream>
+#include <string>
+#include "velox/common/base/BitUtil.h"
+#include "velox/common/base/Exceptions.h"
+#include "velox/type/StringView.h"
+
+#pragma once
+
+#define BUILD_INT128(X, Y) ((int128_t)(X) << 64 | Y)
+
+namespace facebook::velox {
+
+using int128_t = __int128_t;
+
+struct LongDecimal {
```

Comment:
Better have a test somewhere to assert `static_assert(Buffer::is_pod_like_v<LongDecimal>)`.

Reply:
The following code that allocates memory has this check in place. Or else, the code wouldn't compile.

```
template <TypeKind kind>
static VectorPtr createEmpty(
    vector_size_t size,
    velox::memory::MemoryPool* pool,
    const TypePtr& type) {
  using T = typename TypeTraits<kind>::NativeType;
  BufferPtr values = AlignedBuffer::allocate<T>(size, pool);
```

This createEmpty template applies to all TypeKinds.

Reply:
There is a place where it is initialized differently depends on whether it is POD like.

Reply:
Yes, allocate<T> indeed checks for is_pod_like. The above code makes a call to allocate<>, that's why I included here.